src/Money.php

```diff
@@ -204,18 +185,18 @@ public function getCurrency()
      * the sum of this and an other Money object.
      *
      * @param Money[] $addends
-     *
-     * @return Money
      */
-    public function add(Money ...$addends)
+    public function add(Money ...$addends): Money
     {
         $amount = $this->amount;
-        $calculator = $this->getCalculator();
 
         foreach ($addends as $addend) {
-            $this->assertSameCurrency($addend);
+            // Note: non-strict equality is intentional here, since `Currency` is `final` and reliable.
+            if ($this->currency != $addend->currency) {
```

Comment:
assertions have been inlined - kinda needed for these variadic `O(n)` operations.